libs/contrib/Data/List/Elem/Extra.idr

```diff
@@ -44,3 +44,15 @@ notElemAppRight : (ys, xs : List a)
                -> (prf : Not (Elem x (xs ++ ys)))
                -> Not (Elem x ys)
 notElemAppRight ys xs prf = prf . elemAppRight xs ys
+
+||| Proof that if an element belongs to concatenetion of lists xs ++ ys it belongs either to xs of ys
+public export
+hereOrThereConcat : (xs: List a)
+                  -> (ys: List a)
+                  -> (e `Elem` (xs ++ ys))
+                  -> Either (e `Elem` xs) (e `Elem` ys)
+hereOrThereConcat [] ys pos = Right pos
+hereOrThereConcat (e :: xs) ys Here = Left Here
+hereOrThereConcat (x :: xs) ys (There pos) = case hereOrThereConcat xs ys pos of
+                                              (Left pos) => Left $ There pos
+                                              (Right pos) => Right pos
```

Comment:
Isn't this a duplicate of
```
elemAppLorR : (xs, ys : List a)
           -> (prf : Elem k (xs ++ ys))
           -> Either (Elem k xs) (Elem k ys)
```
a few lines above?

Reply:
It is... Thanks for spotting this.